src/internal/connector/onedrive/data_collections_test.go

```diff
@@ -97,13 +108,13 @@ func (suite *DataCollectionsUnitSuite) TestMigrationCollections() {
 
 				for i, cm := range test.expectMigration {
 					if cm.full == fp && cm.prev == pp {
-						migrs[i] = true
+						migrs[i] = cm
 					}
 				}
```

Comment:
we already did a length check above, we don't need to be picky here. We can just do something like
```go
migrs[i] = migr{full: fp, prev: pp}
```

The contains check below will just ensure we have at least what we expect